data/scenario-test.cfg

```diff
@@ -1397,7 +1397,24 @@ My best advancement costs $next_cost gold and I’m $experience|% there."
             # The file is not in the repository to allow adding custom stuff to it.
             [command]
                 [lua]
-                    code=<< wesnoth.dofile("lua/code.lua").main() >>
+                    code=<<
+                      local success, codeTable =
+                        pcall(wesnoth.dofile, "lua/code.lua")
+                      if success then
+                        if type(codeTable) == 'table' and
+                           type(codeTable.main) == 'function'
+                        then
+                          codeTable.main()
+                        else
+                          wesnoth.message(
+                            "code.lua must return a table with a main function.")
+                        end
+                      else
+                        wesnoth.message(
+                          codeTable ..
+                          " (To use this command, create a file 'data/lua/code.lua' in the wesnoth source directory that returns a table with a main function.)")
```

Comment:
This should probably be all on one line. I could understand booting just the string to a new line, since it's long, but there's no reason to start a new line on `codetable ..`